aws/components/directory/setup.ftl

```diff
@@ -0,0 +1,221 @@
+[#ftl]
+[#macro aws_directory_cf_deployment_generationcontract occurrence ]
+    [@addDefaultGenerationContract subsets=[ "template", "epilogue" ] /]
+[/#macro]
+
+[#macro aws_directory_cf_deployment occurrence ]
+    [@debug message="Entering" context=occurrence enabled=false /]
+
+    [#-- Component State helpers --]
+    [#local core = occurrence.Core ]
+    [#local solution = occurrence.Configuration.Solution ]
+    [#local resources = occurrence.State.Resources ]
+
+    [#-- Baseline component lookup --]
+    [#local baselineLinks = getBaselineLinks(occurrence, [ "Encryption" ] )]
+    [#local baselineComponentIds = getBaselineComponentIds(baselineLinks)]
+    [#local cmkKeyId = baselineComponentIds["Encryption"]!"" ]
+
+    [#-- Network Lookup --]
+    [#local networkLink = getOccurrenceNetwork(occurrence).Link!{} ]
+    [#local networkLinkTarget = getLinkTarget(occurrence, networkLink ) ]
+    [#if ! networkLinkTarget?has_content ]
+        [@fatal message="Network could not be found" context=networkLink /]
+        [#return]
+    [/#if]
+    [#local networkConfiguration = networkLinkTarget.Configuration.Solution]
+    [#local networkResources = networkLinkTarget.State.Resources ]
+    [#local vpcId = networkResources["vpc"].Id ]
+
+    [#local dnsPorts = [ 
+        "dns-tcp", "dns-tcp",
+        "globalcatalog",
+        "kerebosauth88-tcp", "kerebosauth88-udp", "kerebosauth464-tcp", "kerebosauth464-udp",
+        "ldap-tcp", "ldap-udp", "ldaps",
+        "netlogin-tcp", "netlogin-udp", 
+        "ntp",
+        "rpc", "ephemeralrpctcp", "ephemeralrpcudp",
+        "rsync",
+        "smb-tcp", "smb-udp", 
+        "anyicmp"
+    ]]
+
+    [#-- Resources and base configuration --]
+    [#local dsId = resources["directory"].Id ]
+    [#local fqdName = resources["directory"].Name ]
+    [#local dsShortName = resources["directory"].ShortName ]
+    [#local securityGroupId = resources["sg"].Id ]
+    [#local securityGroupName = resources["sg"].Name ]
+
+    [#local engine = solution.Engine]
+    [#local enableSSO = solution["aws:EnableSSO"]]
+
+    [#switch engine]
+        [#case "ActiveDirectory"]
+            [#local type = "MicrosoftAD"]
+            [#local size = (solution.Size == "Standard")?then("Standard","Enterprise")]
```

Comment:
This should probably be handled on the component configuration rather than within the setup. 

I can see they are different between the different implementations so we can either make our own values to align with it or specific values.

So on the core provider we either have values as small medium large, and then translate them in the backend or make sizes vendor specific.

Reply:
Made it Small and Large from the feedback on engine-core